include/pmacc/random/methods/XorMin.hpp

```diff
@@ -113,6 +113,19 @@ namespace methods
             return state.v[ 4 ] + state.d;
         }
 
+        DINLINE uint64_t
+        get64Bits(
+            T_Acc const & acc,
+            StateType& state
+        ) const
+        {
+            // two 32bit values are packed into a 64bit value
+            uint64_t result = get32Bits( acc, state);
+            result <<= 31;
```

Comment:
First of all, shouldn't this be a 32 for unsigned integers?

I think this only works for 32 bits - used on a 64 bit unsigned int it fails (for me) resulting in loss of all bits beyond the 32 bits.

Example code:
```c
#include<stdio.h>
#include<stdlib.h>
#include <string.h>

int main()
{
  unsigned int init = 0;
  unsigned int init2 = 1;
  printf("binaer: %016x\n",  init);

  init = ~init;
  printf("binaer: %016x\n",  init);

  long unsigned int result = init;
  printf("long unsigned int:  %2d bit\n", sizeof(result)*8);
  printf("binaer: %016x\n",  result);

  unsigned int i = 0;
  for(i=0; i<34; i++)
    printf("%d: %016x\n", i, result<<i);

  result <<= 31;
  printf("binaer: %016x\n",  result);

  result ^= init2;
  printf("binaer: %016x\n",  result);
  return 0;
}
```

results in:
```
binaer: 0000000000000000
binaer: 00000000ffffffff
long unsigned int:  64 bit
binaer: 00000000ffffffff
0: 00000000ffffffff
1: 00000000fffffffe
2: 00000000fffffffc
3: 00000000fffffff8
4: 00000000fffffff0
5: 00000000ffffffe0
6: 00000000ffffffc0
7: 00000000ffffff80
8: 00000000ffffff00
9: 00000000fffffe00
10: 00000000fffffc00
11: 00000000fffff800
12: 00000000fffff000
13: 00000000ffffe000
14: 00000000ffffc000
15: 00000000ffff8000
16: 00000000ffff0000
17: 00000000fffe0000
18: 00000000fffc0000
19: 00000000fff80000
20: 00000000fff00000
21: 00000000ffe00000
22: 00000000ffc00000
23: 00000000ff800000
24: 00000000ff000000
25: 00000000fe000000
26: 00000000fc000000
27: 00000000f8000000
28: 00000000f0000000
29: 00000000e0000000
30: 00000000c0000000
31: 0000000080000000
32: 0000000000000000
33: 0000000000000000
binaer: 0000000080000000
binaer: 0000000080000001
```

Reply:
yes my shift is wrong. I will fix it and validate the rang again.

Reply:
*offline* discussion with @psychocoderHPC: it should be shifted by 32 bits (occurs in several places)